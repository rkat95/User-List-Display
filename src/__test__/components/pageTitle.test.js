import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import PageTitle from "../../components/PageTitle/pageTitle";
import renderer from "react-test-renderer";
import ReorderIcon from "@mui/icons-material/Reorder";

afterEach(cleanup);

function setHeaderIcon() {
  return;
}

const data = {
  title: "users_display_title",
  subtitle: "users_display_subtitle",
  icons: [
    { icon: ReorderIcon, clickParam: "tableView", clickFn: setHeaderIcon },
  ],
};

describe("Page Title tests", () => {
  it("should render title", () => {
    render(<PageTitle data={data} />);
    const container = screen.getByTestId("page-title");
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent("users_display_title");
  });

  it("should render title div", () => {
    render(<PageTitle data={data} />);
    const container = screen.getByTestId("page-subtitle");
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent("users_display_subtitle");
  });

  it("should match renderer", () => {
    const tree = renderer.create(<PageTitle data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
