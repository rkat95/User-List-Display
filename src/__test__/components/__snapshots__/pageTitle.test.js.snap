// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Title tests should match renderer 1`] = `
<div
  className="page-header-container"
>
  <div
    className="top-container"
  >
    <div
      className="title-container"
    >
      <div
        className="page-title"
        data-testid="page-title"
      >
        users_display_title
      </div>
      <div
        className="page-subtitle"
        data-testid="page-subtitle"
      >
        users_display_subtitle
      </div>
    </div>
    <div
      className="icons-container"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium action-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="icon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M3 15h18v-2H3v2zm0 4h18v-2H3v2zm0-8h18V9H3v2zm0-6v2h18V5H3z"
        />
      </svg>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth title-divider css-9mgopn-MuiDivider-root"
  />
</div>
`;
